src/webgl/material.js

```diff
@@ -343,4 +323,34 @@ p5.RendererGL.prototype._applyColorBlend = function(v1,v2,v3,a){
   return colors;
 };
 
+
+p5.prototype.loadShader = function(fragShader, vertShader) {
+  var loadedShader = new p5.Shader();
```

Comment:
Ok this makes sense.  I think we should keep your implementation as is.  Perhaps a very small change should be to rename the loadedShader variable as shaderProgram, since technically the shader isn't loaded until the frag and vert sources have finished loading asynchronously.